compiler/luci-micro/CMakeLists.txt

```diff
@@ -0,0 +1,95 @@
+
```

Comment:
I was able to build interpreter using this configuration, but it need some refactoring for proper integration within rest of the build.

This is how I see use cases for this CMake:
- Automated build for x86 micro profile for easy integration with CI and automated tests.
- Automated build for arm architecture to ckeck if interpreter build without errors.
- Manually triggered build of test program (build with mbed os). This could be useful for testing/benchmarking purposes.

To achieve integration I propose:
- Remove required from this file, it is needed for mbed-os build only (mbed-os has this required inside already)
- Use existing targets, instead of including dependency directories manually
- Use existing externals infrastructure instead of downloading dependencies in this file.
- Write our own cmake toolchain file for mbed-os instead of patching existing one.
- Remove mbed-os patch, if we use our own toolchain file it is not needed (other changes look redundant in case we don't have spaces in paths).
- Optionally. Add Dockerfile for manual ARM build, because mbed os requires new cmake and python, not available on CI and many developer machines.